cryptographic_estimators/UOVEstimator/uov_problem.py

```diff
@@ -0,0 +1,87 @@
+# ****************************************************************************
+# Copyright 2023 Technology Innovation Institute
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# ****************************************************************************
+
+
+from ..base_problem import BaseProblem
+from sage.arith.misc import is_prime_power
+
+
+class UOVProblem(BaseProblem):
+    """
+    Construct an instance of UOVProblem.
+
+    INPUT: 
+        
+        - ``n`` -- number of variables
+        - ``m`` -- number of polynomials
+        - ``q`` -- order of the finite field (default: None)
+
+    """
+
+    def __init__(self, n: int, m: int, q:int, **kwargs):
+        if n < 1:
+            raise ValueError("n must be >= 1")
+
+        if m < 1:
+            raise ValueError("m must be >= 1")
+
+        if q is not None and not is_prime_power(q):
+            raise ValueError("q must be a prime power")
+        
+        super().__init__(**kwargs)
+        self.parameters["UOV_NUMBER_VARIABLES"] = n
+        self.parameters["UOV_NUMBER_POLYNOMIALS"] = m
+        self.parameters["UOV_FIELD_SIZE"] = q
+
+    def to_bitcomplexity_time(self, basic_operations: float):
+        """
+        Return the bit-complexity corresponding to a certain amount of basic_operations
+
+        INPUT:
+
+        - ``basic_operations`` -- Number of basic operations (logarithmic)
+
+        """
+        return basic_operations
```

Comment:
Please implement method.

For signature schemes this function must input `basic_operations` and `number_of_hashes`; both should be `float`.

The description of the number of hashes should be 
```   - ``number_of_hashes`` -- Number of hashes  (logarithmic) (default: 0)``` 

This parameter indicates the number of hashes the algorithm has to compute during its execution. 

The bit-complexity output of this algorithm should be the sum of the bit_complexity comming from the numer of `basic_operations` and the number_of_hashes`.  Transform the number of `basic_operations` into bit-complexity should be done as in the MQProblem class. The bit-complexity commming from the number of hashes should be `number_of_hashes` + `cost_hash`, where `cost_hash` is the bit-complexity (in logarithm) of computing one hash, this value should be an attibute of the UOVProblem class and it should have a defult value of `17`.